src/tracer.js

```diff
@@ -237,8 +240,7 @@ export default class Tracer {
             ctx.baggage = parent.baggage;
 
             parent.finalizeSampling();
-            this.processDeferredSampling(ctx, operationName, internalTags);
-            ctx.finalizeSampling();
+            this.handleUpsampling(parent, ctx, operationName, internalTags);
```

Comment:
`handleUpsampling` does not call `ctx.finalizeSampling`; why would the first operationName change be disregarded?
(This case seems to be missing a test, let me add it)